plugins/quay/src/api/index.test.ts

```diff
@@ -0,0 +1,124 @@
+
+  const getConfigApi = (getOptionalStringFn: any) => ({
+    has: jest.fn(),
+    keys: jest.fn(),
+    get: jest.fn(),
+    getBoolean: jest.fn(),
+    getConfig: jest.fn(),
+    getConfigArray: jest.fn(),
+    getNumber: jest.fn(),
+    getString: jest.fn(),
+    getStringArray: jest.fn(),
+    getOptional: jest.fn(),
+    getOptionalStringArray: jest.fn(),
+    getOptionalBoolean: jest.fn(),
+    getOptionalConfig: jest.fn(),
+    getOptionalConfigArray: jest.fn(),
+    getOptionalNumber: jest.fn(),
+    getOptionalString: getOptionalStringFn,
+  });
+
+  beforeEach(() => {
+    quayApi = new QuayApiClient({
+      configApi: getConfigApi(() => {
+        return '/quay';
+      }),
+      discoveryApi: UrlPatternDiscovery.compile('https://localhost:5050'),
+    });
+  });
+
+  it('should use a correct default proxy path', async () => {
+    quayApi = new QuayApiClient({
+      configApi: getConfigApi(jest.fn()),
+      discoveryApi: UrlPatternDiscovery.compile('https://localhost:5050'),
+    });
+
+    const result = await quayApi.getTags('foo', 'bar');
+
+    expect(result).toEqual(
+      require(`${__dirname}/../../fixtures/tags/foo_page1.json`),
+    );
+  });
+
+  it('should throw an error when the response is not ok', async () => {
+    await expect(quayApi.getTags('not', 'found')).rejects.toEqual(
+      new Error('failed to fetch data, status 404: Not Found'),
+    );
+  });
+
+  describe('getTags', () => {
+    it('should correctly get tags without optional arguments', async () => {
+      const result = await quayApi.getTags('foo', 'bar');
+
+      expect(result).toEqual(
+        require(`${__dirname}/../../fixtures/tags/foo_page1.json`),
+      );
+    });
+
+    it('should correctly get tags with a limit', async () => {
+      const result = await quayApi.getTags('foo', 'bar', undefined, 1);
+
+      expect(result).toEqual(
+        require(`${__dirname}/../../fixtures/tags/foo_limit.json`),
+      );
+    });
+    it('should correctly get tags with a page number', async () => {
+      const result = await quayApi.getTags('foo', 'bar', 2);
+
+      expect(result).toEqual(
+        require(`${__dirname}/../../fixtures/tags/foo_page2.json`),
+      );
+    });
+  });
+
+  describe('getManifestByDigest', () => {
+    it('should correctly get the manifest using its digest', async () => {
+      const manifest = require(`${__dirname}/../../fixtures/manifests/foo.json`);
+      const result = await quayApi.getManifestByDigest(
+        'foo',
+        'bar',
+        manifest.digest,
+      );
+
+      expect(result).toEqual(manifest);
+    });
+  });
+
+  describe('getLabels', () => {
+    it('should correctly get the labels using the manifest digest', async () => {
+      const manifest = require(`${__dirname}/../../fixtures/manifests/bar.json`);
+      const result = await quayApi.getLabels('foo', 'bar', manifest.digest);
+
+      expect(result).toEqual(
+        require(`${__dirname}/../../fixtures/labels/foo.json`),
```

Comment:
I looked at the fixtures directory and it seems like only this test is using them + the handlers. You should move `${PROJECT_DIR}/fixtures` to instead be located at `${PROJECT_DIR}/src/api/fixtures`. 
If we decide to add more tests in other directories in the future which reference these test files, we can think about moving it to a higher directory then. 
This way, we aren't emphasizing the importance of test data in our project.

Reply:
I agree with that, moved the fixtures and the handlers to the api folder.